### PR TITLE
Added GC functionality in test_Mbuckets_with_Nobjects.py file

### PR DESCRIPTION
**Pull request by udaysk23**
Added gc functionality in test_Mbuckets_with_Nobjects.py file for multiple usage and removed test_gc_list_verification.py since covering same task in test_Mbuckets_with_Nobjects.py

Log link for this run is as following:
http://pastebin.test.redhat.com/897465

### CODE REVIEW

**Comment by rakeshgm** [comment]
More yamls can be derived from this. 
- multipart, versioning and also compression, too.

File: rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py [hunk 1]
```diff
@@ -116,6 +117,38 @@ def test_exec(config):
...
                 log.info('RGW service restarted')
+        if config.gc_verification is True:
+            config.bluestore_block_size = 1549267441664
+            config.rgw_gc_max_queue_size = 367788
+            config.rgw_gc_processor_max_time = 3600
+            config.rgw_gc_max_concurrent_io = 10
+            config.rgw_gc_max_trim_chunk = 32
+            config.rgw_objexp_gc_interval = 10
+            config.rgw_gc_obj_min_wait = 10
+            config.rgw_gc_processor_period = 10
```
Comment (rakeshgm) [inline rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:128]: Can these be taken from yaml config, if the values are present you can use these as default values

Comment (udaysk23) [reply rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:128]: Instead of taking it from yaml I would like to hard-code these values in script itself.

Comment (rakeshgm) [reply rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py]: That was not the intent of the my comment. Hardcodimg values in the code is always bad.
If the values are present in the config, then use them or use the values in the py file. A simple dictonary get() should help you to achieve this.

Comment (udaysk23) [reply rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py]: gc values are now handled from yaml file

Comment (udaysk23) [reply rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py]: script takes ceph conf values as a dictionary from yaml file now

File: rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py [hunk 2]
```diff
@@ -253,6 +285,14 @@ def test_exec(config):
             else:
                 log.info('RGW service restarted')
 
+        if config.gc_verification is True:
+            op = utils.exec_shell_cmd('radosgw-admin gc list')
+            # ERROR: failed to list objs: (22) Invalid argument
```
Comment (rakeshgm) [inline rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:290]: Can you add more information in this comment.

File: rgw/v2/tests/s3_swift/configs/test_gc_list.yaml [hunk 3]
```diff
@@ -1,9 +1,31 @@
...
+      create_bucket: true
+      create_object: true
+      download_object: false
+      delete_bucket_object: true
+      sharding:
+        enable: false
+        max_shards: 0
+      compression:
+        enable: false
+        type: zlib
```
Comment (rakeshgm) [inline rgw/v2/tests/s3_swift/configs/test_gc_list.yaml:31]: add a new line here.

**Comment by rakeshgm** [comment]
Changes looks good now.  Can you aslo add other configuration yamls. for multipart, versioning, life-cycle. 
basically what ever combinations with this file.  And also update run-all.sh too.

**Comment by rakeshgm** [comment]
squash commits to reflect proper changes.

File: rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py [hunk 4]
```diff
@@ -116,6 +117,37 @@ def test_exec(config):
...
+            srv_restarted = rgw_service.restart()
+            time.sleep(30)
+            if srv_restarted is False:
+                raise TestExecError("RGW service restart failed")
+            else:
+                log.info('RGW service restarted')
+            # Delete gc queue
+            pool_name = utils.exec_shell_cmd('ceph df |awk \'{ print $1 }\'| grep rgw.log')
+            pool_name = pool_name.replace("\n", "")
+            utils.exec_shell_cmd('for q in {0..31}; do rados rm gc.${q} -p %s -N gc; done' % pool_name)
```
Comment (rakeshgm) [inline rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:149]: your executing a for loop here. how about adding it in python and not use shell cmd here. there is room for improvement. this is not a ceph-command i think.

**Comment by udaysk23** [comment]
> Changes looks good now. Can you aslo add other configuration yamls. for multipart, versioning, life-cycle.
> basically what ever combinations with this file. And also update run-all.sh too.

Hi Rakesh,

Since we need huge number of IOs like in millions, we cant add this in run-all.sh file.

**Comment by rakeshgm** [comment]
> Changes looks good now. Can you aslo add other configuration yamls. for multipart, versioning, life-cycle.
> basically what ever combinations with this file. And also update run-all.sh too.

What about versioning tests. Can you add those configs here as well

**Comment by rakeshgm** [comment]
@udaysk23 rebase needed.

File: rgw/v2/tests/s3_swift/configs/test_gc_list.yaml [hunk 5]
```diff
@@ -1,9 +1,32 @@
 # BZ id : 1837206
-# script: test_gc_list_verification.py
+# script: test_gc_list_verification.
```
Comment (rakeshgm) [inline rgw/v2/tests/s3_swift/configs/test_gc_list.yaml:2]: the script name is wrong here,

File: rgw/v2/tests/s3_swift/configs/test_gc_list_multipart.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# BZ id : 1837206
+# script: test_gc_list_verification.
```
Comment (rakeshgm) [inline rgw/v2/tests/s3_swift/configs/test_gc_list_multipart.yaml:2]: the script name is wrong here.

**Comment by udaysk23** [comment]
Following is the log location of successful run

http://pastebin.test.redhat.com/907035

**Comment by rakeshgm** [comment]
@udaysk23 squash commits.

**Comment by viduship** [comment]
Hi Uday,

It would be even better if we also add the doc string in the Mbuckets_Nobjects.py for GC coverage as part of this PR.

**Review by rakeshgm — APPROVED** [review]

**Pull request merged**